### PR TITLE
Dry up additional spots of setting cluster settings in internal cluster tests

### PR DESCRIPTION
**Pull request by original-brownbear**
Follow up to #94213 dealing with the remaining spots. This also moves some use of deprecated transient settings to persisted settings and deprecates the request builder methods for transient settings to hopefully prevent more use of transients.

### CODE REVIEW

**Review by williamrandolph — CHANGES_REQUESTED** [review]
Looks good overall, but there's one spot where I think this PR might have dropped an update. Let me know if there was a reason for it that I missed!

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java [hunk 1]
```diff
@@ -104,28 +104,18 @@ public class ClassificationIT extends MlNativeDataFrameAnalyticsIntegTestCase {
...
-            .cluster()
-            .prepareUpdateSettings()
-            .setTransientSettings(
-                Settings.builder()
-                    .putNull("logger.org.elasticsearch.xpack.ml.process")
-                    .putNull("logger.org.elasticsearch.xpack.ml.dataframe")
-            )
-            .get();
+        updateClusterSettings(
+            Settings.builder().putNull("logger.org.elasticsearch.xpack.ml.process").putNull("logger.org.elasticsearch.xpack.ml.dataframe")
```
[the remark this answers is not in the document]
Comment (original-brownbear) [reply x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java:117]: We're still setting it to null, the formatter just moved both setting to a single line but functionally nothing changed?

[the remark this answers is not in the document]
Comment (williamrandolph) [reply x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java:117]: Whoops! I was deceived by my diff window and didn't see the whole line! Apologies, and now I will happily approve.

**Review by williamrandolph — APPROVED** [review]
Looks good to me!

**Comment by original-brownbear** [comment]
Thanks William!

**Pull request merged**